tasks/collect-slack-notification-params/tests/test-collect-slack-notification-params-no-channel.yaml

```diff
@@ -0,0 +1,80 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-collect-slack-notification-params-no-channel
+spec:
+  description: |
+    Run the collect-slack-notification-params task and verify the results
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-values
+            image: quay.io/hacbs-release/release-utils:48a1cd6c939eda25382472135be4afe2c9b4b703
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > $(workspaces.data.path)/data.json << EOF
+              {
+              }
+              EOF
+              cat > $(workspaces.data.path)/snapshot.json << EOF
+              {
+                "application": "myapp",
+                "components": [
+                  {
+                    "name": "comp",
+                    "containerImage": "registry.io/image:tag"
+                  }
+                ]
+              }
+              EOF
+    - name: run-task
+      taskRef:
+        name: collect-slack-notification-params
+      params:
+        - name: release
+          value: "dev-team/release-123"
+        - name: releaseplanadmission
+          value: "managed-team/rpa-456"
+        - name: releasepipelinename
+          value: "$(context.pipelineRun.name)"
+        - name: snapshotPath
+          value: snapshot.json
+        - name: dataPath
+          value: data.json
+        - name: hacUrl
+          value: "https://my.website.ca"
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - setup
+    - name: check-result
+      params:
+        - name: message
+          value: $(tasks.run-task.results.message)
+      taskSpec:
+        params:
+          - name: message
+        steps:
+          - name: check-result
+            image: quay.io/hacbs-release/release-utils:48a1cd6c939eda25382472135be4afe2c9b4b703
+            env:
+              - name: "MESSAGE"
+                value: '$(params.message)'
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              echo "message: ${MESSAGE/$'\n'}"
+              test "x${MESSAGE/$'\n'}x" == "xx"
```

Comment:
huh? So you are testing the `${MESSAGE/$'\n'} == ""`?

Reply:
Yeah, this reminds me of a technique that was used in the past if you needed to account for empty strings.

For example, this will fail:
```sh
AA="" test $AA = ""
bash: test: =: unary operator expected
```

But it's ok once you wrap it in `""`.
```sh
AA="" test "$AA" = ""
```

I think there was a time when even the latter one wouldn't work and you'd need to add something like `x` to both sides. It was probably some very old versions of bash that would just ignore an empty parameter when evaluating the arguments.

Reply:
Also, since you fixed the empty result generation to avoid the empty line, you will no longer have to remove it here.